core/services/payment_methods/forms/PayPalSettingsForm.php

```diff
@@ -0,0 +1,182 @@
+<?php
+namespace EventEspresso\core\services\payment_methods\forms;
+
+use EE_Payment_Method_Form;
+use EE_Text_Input;
+
+/**
+ * Class PayPalSettingsForm
+ *
+ * Parent class for PayPal Payment methods that use the username, password, signature credentials.
+ *
+ * Validates those credentials on save to verify they work.
+ *
+ * @package     Event Espresso
+ * @author         Mike Nelson
+ * @since         $VID:$
+ *
+ */
+class PayPalSettingsForm extends EE_Payment_Method_Form
+{
+    /**
+     * @var string of HTML being the help tab link
+     */
+    protected $helpTabLink;
+
+    public function __construct(array $options_array = array(), $help_tab_link)
+    {
+        $this->helpTabLink = $help_tab_link;
+        $options_array = array_replace_recursive(
+            array(
+            'extra_meta_inputs' => array(
+                'api_username' => new EE_Text_Input(
+                    array(
+                        'html_label_text' => sprintf(
+                            esc_html__('API Username %s', 'event_espresso'),
+                            $help_tab_link
+                        ),
+                        'required'        => true,
+                    )
+                ),
+                'api_password' => new EE_Text_Input(
+                    array(
+                        'html_label_text' => sprintf(
+                            esc_html__('API Password %s', 'event_espresso'),
+                            $help_tab_link
+                        ),
+                        'required'        => true,
+                    )
+                ),
+                'api_signature' => new EE_Text_Input(
+                    array(
+                        'html_label_text' => sprintf(
+                            esc_html__('API Signature %s', 'event_espresso'),
+                            $help_tab_link
+                        ),
+                        'required'        => true,
+                    )
+                ),
+            )
+            )
+        );
```

Comment:
Are we missing a second parameter (`$options_array`) for the `array_replace_recursive()` here ?

Reply:
errrmm ya totally